### PR TITLE
Prevent RequestManager from refreshing page without changes

### PR DESCRIPTION
**Pull request by apicellaj**
Proposed fix for issue #13 that should avoid false negatives. Seems to work okay for me but I'm still not completely familiar with the request manager.

### CODE REVIEW

File: src/Platform_Core/org/lobobrowser/security/PermissionSystem.java [hunk 1]
```diff
@@ -86,6 +86,13 @@ public void dump() {
     boards.get(0).dump();
     getLastBoard().dump();
   }
+  
+  public String dumpPermissions() {
+	  StringBuilder permissions = new StringBuilder();
+	  permissions.append(boards.get(0).dumpString());
+	  permissions.append(getLastBoard().dumpString());
```
Comment (hrj) [inline src/Platform_Core/org/lobobrowser/security/PermissionSystem.java]: * Please make it iterate over all boards, not just first and last one.
* Also, I think `getPermissionsAsString()` would be a better function name. I usually use `dump` for debugging to stdout.

File: src/Platform_Core/org/lobobrowser/security/PermissionSystem.java [hunk 2]
```diff
@@ -141,6 +148,15 @@ public void dump() {
         System.out.println("");
       });
     }
+    
+    public String dumpString() {
+    	StringBuilder permissionState = new StringBuilder();
```
Comment (hrj) [inline src/Platform_Core/org/lobobrowser/security/PermissionSystem.java]: Minor: indentation is wrong.

File: src/Platform_Core/org/lobobrowser/security/PermissionSystem.java [hunk 3]
```diff
@@ -203,6 +219,18 @@ public void dump() {
...
       }
+      
+      public String dumpString() {
+    	  StringBuilder rowPermissionState = new StringBuilder();
+          Arrays.stream(requestCells).forEach(c -> {
+            final PermissionResult permissionResult = c.getEffectivePermission();
+            final String permStr = permissionResult.permission.toString().substring(0, 1);
+            final String permFmtStr = permissionResult.isDefault ? " " + permStr + " " : "[" + permStr + "]";
+            rowPermissionState.append(permFmtStr);
+            System.out.print(String.format(" %s%d", permFmtStr, c.parentCells.size()));
```
Comment (hrj) [inline src/Platform_Core/org/lobobrowser/security/PermissionSystem.java]: Same here: indentation is wrong in this function.

Also, can you remove the System.out.print() call? That was for debugging right?

**Comment by hrj** [comment]
Thanks for the PR! I only have minor comments; will merge after you address them.

----
Converting to string is hackish, but I can't think of a better method that is simple and straightforward. :shipit:

**Comment by apicellaj** [comment]
Thanks for the feedback! I agree that converting to string is hackish, but I couldn't figure out a cleaner way to do it. I'm definitely open to other suggestions.

**Comment by hrj** [comment]
@apicellaj I noticed that you tried to fix the indentation but it still seems broken. It might be because of mixed tabs and spaces. If you are using Eclipse, have you seen the [how to setup code formatter](https://github.com/UprootLabs/gngr/wiki/Developer:Setup#how-to-import-the-code-formatter-settings) section of the wiki? For other IDEs / editors, something similar might need to be setup (2 spaces for indentation, no tabs).

**Comment by apicellaj** [comment]
@hrj I'm using Eclipse and was previously using the formatting you mentioned but I might have switched things while setting project specific settings for a different project.

I think I fixed everything by running the format again. In addition, an unrelated line of code seems to have been split into two lines by the formatter.

**Comment by hrj** [comment]
Thanks @apicellaj !

> In addition, an unrelated line of code seems to have been split into two lines by the formatter.

Yeah, there is some co-lateral damage sometimes. If you don't like a change, you can selectively commit using `git add -p` or similar.

It's good that you noticed it; keeping changes small helps reduce merge conflicts (I have a couple of unstable branches in gngr that I need to keep in sync with master).

**Pull request merged**